test/blackbox-tests/test-cases/default_cache.t

```diff
@@ -0,0 +1,32 @@
+The dune cache should be enabled by default
+  $ echo "(lang dune 3.16)" > dune-project
+
+  $ cat > dune << EOF
+  > (library
+  >  (name foo))
+  > EOF
+
+  $ cat > foo.ml << EOF
+  > let f x y = x + y
+  > EOF
+
+Set up cache directory
+  $ rm -rf /tmp/dune_test_cache
+  $ mkdir /tmp/dune_test_cache
```

Comment:
Do not create files outside the test directory. $(pwd)/dune_test_cache` is a better choice.